core/models/provider.py

```diff
@@ -203,7 +203,23 @@ def get_platform_name(self):
     def get_type_name(self):
         return self.type.name
 
+    def is_current(self):
+        """
+        Is current when:
+        - No end date present
+        - End date is earlier than 'now'
+        """
+        if not self.end_date:
+            return True
+        now = timezone.now()
+        return self.end_date < now
```

Comment:
This logic can be cleaned up, return a single boolean expression.

I think this should be:
```
return not self.end_date || self.end_date > now
```